src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -14,17 +14,19 @@ function fill_open_boundary_regions!(field, boundary_conditions, indices, loc, g
 
     # gets `open_fill`, the function which fills open boundaries at `loc`, as well as `regular_fill`
     # which is the function which fills non-open boundaries at `loc` which informs `fill_halo_size` 
-    open_fill, regular_fill = get_open_halo_filling_functions(loc) 
-
-    # Computing the size of the fill halo kernel. We use the `regular_fill` function to
-    # calculate the index range. Remember: windowed fields require the halos
-    # to be computed only on a part of the boundary so we need to offset the kernel indices 
-    size   = fill_halo_size(field, regular_fill, indices, boundary_conditions, loc, grid)
-    offset = fill_halo_offset(size, regular_fill, indices)
-    params = KernelParameters(size, offset)
+    fill_halo_function = get_open_halo_filling_functions(loc) 
+
+    if !isnothing(fill_halo_function)
+        # Computing the parameters of the fill halo kernel. 
+        # Remember: windowed fields require the halos to be computed only on a part of the boundary 
+        # so we need to offset the kernel indices 
+        size   = fill_halo_size(field, fill_halo_function, indices, boundary_conditions, loc, grid)
+        offset = fill_halo_offset(size, fill_halo_function, indices)
+        params = KernelParameters(size, offset)
+        
+        launch!(arch, grid, params, fill_halo_function, field, left_bc, right_bc, loc, grid, args)
+    end
```

Comment:
Why not replace this with a function that fills the halos?

Reply:
Yeah, I changed this PR to reuse the implementation we already have in `fill_halo_regions.jl`. I found nothing special for open boundary conditions to warrant writing a new function.